llvm/lib/Target/AMDGPU/SILoadStoreOptimizer.cpp

```diff
@@ -2571,3 +2580,24 @@ bool SILoadStoreOptimizer::runOnMachineFunction(MachineFunction &MF) {
 
   return Modified;
 }
+
+PreservedAnalyses
+SILoadStoreOptimizerPass::run(MachineFunction &MF,
+                              MachineFunctionAnalysisManager &MFAM) {
+  if (MF.getFunction().hasOptNone())
+    return PreservedAnalyses::all();
+
+  MFPropsModifier _(*this, MF);
```

Comment:
This should probably be before the optnone skip

Reply:
optnone is like `skipFunction` and properties check is a part of the pass, so I thought we should skip the check as well.

Reply:
Yes, but in terms of property changes I would expect the pass to always behave the same way

Reply:
And the legacy pass manager works the other way, so I'll change that.